tests/php/Functional/WC/Payment/Mollie_WC_Payment_Order_Test.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Mollie\WooCommerceTests\Functional\Payment;
+
+use Mollie\Api\Resources\Order;
+use Mollie\WooCommerceTests\TestCase;
+
+use Mollie_WC_Helper_Data;
+use Mollie_WC_Payment_Order;
+use Mollie_WC_Payment_OrderItemsRefunder;
+use WC_Order;
+
+use function Brain\Monkey\Functions\when;
+
+
+class Mollie_WC_Payment_Order_Test extends TestCase
+{
+    /* -----------------------------------------------------------------
+       onWebhookCanceled Tests
+       -------------------------------------------------------------- */
+
+    /**
+     * given onWebhookCanceled is called
+     * when order has status of Completed|Refunded|Cancelled
+     * then status does NOT change and we exit the method.
+     *
+     * @test
+     */
+    public function onWebHookCanceled_Returns_whenFinalStatus(){
+        /*
+        * Setup Stubs
+        */
+        $orderItemsRefunded = $this->buildTesteeMock(
+            Mollie_WC_Payment_OrderItemsRefunder::class,
+            [],
+            []
+        )->getMock();
+
+        $payment = $this->buildTesteeMock(
+            Order::class,
+            [],
+            []
+        )->getMock();
+        $order = $this->buildTesteeMock(
+            WC_Order::class,
+            [],
+            []
+        )->getMock();
+        $dataHelper = $this->buildTesteeMock(
+            Mollie_WC_Helper_Data::class,
+            [],
+            ['hasOrderStatus']
+        )->getMock();
+        $payment_method_title = 'creditcard';
+
+        /*
+        * Setup Testee
+        */
+        $data = 'order';
+        $testee = new Mollie_WC_Payment_Order($orderItemsRefunded, $data);
+
+        /*
+         * Expectations
+         */
+        when('wooCommerceOrderId')
+            ->justReturn(89);
```

Comment:
For unit tests I would not set specific values since you're not interested in the values per se but in the logic, I would suggest to introduce https://github.com/fzaninotto/Faker as a library to create stub values.

So you can simply do `when('wooCommerceOrderId')->justReturn($this->faker->randomNumber(2))`

it's also possible to use php functions for simple values but the library expose methods and properties to generate a wide range of values which is good because you're tests will be shorten and more simple to read.